### PR TITLE
JSDoc, rename Symbol to CodeSymbol

### PR DESCRIPTION
**Pull request by mischnic**
# ↪️ Pull Request

1. Rename Symbol to CodeSymbol (to not clash with the ES6 Symbol) - open to better suggestions
2. Add JSDoc-like docs to `@parcel/types`

Closes https://github.com/parcel-bundler/parcel/issues/3926

I've added comments to the places where I wasn't quite sure.

To upstream https://parcel2-docs.now.sh/plugin-system/api/

### CODE REVIEW

File: packages/core/types/index.js [hunk 1]
```diff
@@ -240,7 +305,11 @@ export type HTTPSOptions = {|
   +key: FilePath,
 |};
 
-// Source locations are 1-based, meaning lines and columns start at 1
+/**
+ * Source locations are 1-based, meaning lines and columns start at 1
+ * @property start inclusive
+ * @property end exclusive, FIXME?
```
Comment (mischnic) [inline packages/core/types/index.js:311]: Is this true?

File: packages/core/types/index.js [hunk 2]
```diff
@@ -268,9 +348,28 @@ export type DependencyOptions = {|
...
 
+/**
+ * A Dependency denotes a connection between two assets \
+ * (likely some effect from the importee is expected - be it a side effect or a value is being imported).
+ *
+ * @property moduleSpecifier E.g. "lodash" in <code>import {add} from "lodash";</code>
+ * @property isAsync Whether the environment can load this natively (FIXME).
+ * @property isEntry Whether this should become a new entry (e.g. no hash).
+ * @property isOptional Whether the importer might expect an error when resolving failed.
+ * @property isURL Whether an URL is expected (rather than the language-specific behaviour).
```
Comment (mischnic) [inline packages/core/types/index.js:362]: Really not sure about async/optional/url

File: packages/core/types/index.js [hunk 3]
```diff
@@ -300,6 +399,30 @@ export type ASTGenerator = {|
   version: string,
 |};
 
+/**
+ * An asset (usually represents one source file).
+ *
+ * @property The file system where the source is located.
+ * @property isIsolated Whether this asset should be put in a separate bundle. VERIFY
+ * @property isInline Whether this asset will/should later be inserted back into the importer.
+ * @property isSplittable FIXME
```
Comment (mischnic) [inline packages/core/types/index.js:408]: Isolated vs splittable?

File: packages/core/types/index.js [hunk 4]
```diff
@@ -352,10 +480,34 @@ export interface MutableAsset extends BaseAsset {
...
+ * @method setResultHash FIXME
+ * @method addIncludedFile FIXME
+ * @method addDevDependency FIXME
+ * @method setWatchGlob FIXME
+ * @method getConfigFrom FIXME
+ * @method getConfig FIXME
+ * @method getPackage FIXME
+ * @method shouldRehydrate FIXME
+ * @method shouldReload FIXME
+ * @method shouldInvalidateOnStartup FIXME
```
Comment (mischnic) [inline packages/core/types/index.js:509]: The whole `Config` class

File: packages/core/types/index.js [hunk 5]
```diff
@@ -468,6 +623,18 @@ export type MultiThreadValidator = {|
 
 export type Validator = DedicatedThreadValidator | MultiThreadValidator;
 
+/**
+ * The methods for a transformer plugin.
+ * @method getConfig (deprecated)
+ * @method loadConfig FIXME
+ * @method preSerializeConfig FIXME
+ * @method postDeserializeConfig FIXME
```
Comment (mischnic) [inline packages/core/types/index.js:631]: Config related transformer methods

File: packages/core/types/index.js [hunk 6]
```diff
@@ -575,12 +763,31 @@ export type CreateBundleOpts =
...
+/**
+ * A Bundle (a collection of assets)
+ * @property hashReference Whether this value is inside <code>filePath</code> it will be replace with the real hash at the end.
+ * @property isEntry Whether this is an entry (e.g. should not be hashed).
+ * @property isInline Whether this bundle should be inlined into the parent bundle(s),
+ * @property isSplittable FIXME
+ * @property filePath The output filespath (if not inline), can contain <code>hashReference</code> before the optimizer ran.
+ *
+ * @method getEntryAssets Assets that run when the bundle is loaded (e.g. runtimes could be added). VERIFY
+ * @method getMainEntry The actual entry (which won't be a runtime), the same as the last entry in <code>getEntryAssets()</code>
```
Comment (mischnic) [inline packages/core/types/index.js:787]: Not sure about `isSplittable` and if I got the entries right

File: packages/core/types/index.js [hunk 7]
```diff
@@ -602,38 +809,41 @@ export interface Bundle {
...
+ * @method getChildBundles Child bundles are Bundles that might be loaded by an asset in the bundle
+ * @method getSiblingBundles See BundleGroup
+ * @method getReferencedBundles Bundles that are referenced (by filename)
+ * @method getDependencies Get the dependencies of the asset
+ * @method getIncomingDependencies Get the dependencies that require the asset
+ * @method resolveExternalDependency Returns undefined if the specified dependency was excluded or wasn't async \
+ * and otherwise the BundleGroup or Asset that the dependency resolves to. VERIFY
+ * @method getDependencyResolution Find out which asset the dependency resolved to.
+ * @method isAssetInAncestorBundles Whether the asset is already included in a compatible (regarding EnvironmentContext) parent bundle.
+ * @method isAssetReferenced Whether the asset is referenced (the "references" edge)FIXME how? url/filename?
```
Comment (mischnic) [inline packages/core/types/index.js:840]: What does the "references" edge signify?

File: packages/core/types/index.js [hunk 8]
```diff
@@ -670,19 +880,60 @@ export interface BundleGraph {
...
+/**
+ * A BundleGraph in the Bundler that can be modified
+ * @method addAssetGraphToBundle Add asset and all child nodes to the bundle VERIFY how??
+ * @method createAssetReference FIXME
+ * @method createBundleGroup Turns an edge (Dependency -> Asset-s) into (Dependency -> BundleGroup -> Asset-s)
+ * @method getDependencyAssets FIXME a dependency can have multiple child nodes?
+ * @method removeAssetGraphFromBundle Remove all "contains" edges from the bundle to the nodes in the asset's subgraph.
+ * @method internalizeAsyncDependency Turns a dependency to a different bundle into a dependency to an asset inside <code>bundle</code>.
+ * @method traverse FIME difference to traverseContents?
+ * @method traverseContents FIXME
```
Comment (mischnic) [inline packages/core/types/index.js:892]: 1. the "references" edge again
2. `getDependencyAssets`?
3. traverse and traverseContents seem to do the same thing?

**Comment by parcel-benchmark** [comment]
## Benchmark Results
<details><summary>Kitchen Sink ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 6.65s | -403.00ms 🚀 |
| Cached | 2.04s | -389.00ms 🚀 |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/legacy/parcel.d5807e82.webp | 102.94kb | +0.00b | 219.00ms | -32.00ms 🚀 |
| dist/modern/parcel.d5807e82.webp | 102.94kb | +0.00b | 219.00ms | -32.00ms 🚀 |
| dist/legacy/kitchen-sink.32bb4226.js | 1.13kb | +0.00b | 1.61s | -263.00ms 🚀 |
| dist/modern/kitchen-sink.280a0cce.js | 1.13kb | +0.00b | 1.61s | -262.00ms 🚀 |
| dist/legacy/index.html | 709.00b | +0.00b | 605.00ms | -80.00ms 🚀 |
| dist/modern/index.html | 709.00b | +0.00b | 604.00ms | -76.00ms 🚀 |
| dist/legacy/styles.afb8e31a.css | 78.00b | +0.00b | 1.61s | -263.00ms 🚀 |
| dist/modern/styles.cda8c91b.css | 78.00b | +0.00b | 1.23s | -298.00ms 🚀 |

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/legacy/parcel.d5807e82.webp | 102.94kb | +0.00b | 183.00ms | -18.00ms 🚀 |
| dist/modern/parcel.d5807e82.webp | 102.94kb | +0.00b | 183.00ms | -18.00ms 🚀 |
| dist/legacy/kitchen-sink.32bb4226.js | 1.13kb | +0.00b | 319.00ms | -71.00ms 🚀 |
| dist/modern/kitchen-sink.280a0cce.js | 1.13kb | +0.00b | 317.00ms | -73.00ms 🚀 |
| dist/legacy/index.html | 709.00b | +0.00b | 321.00ms | -72.00ms 🚀 |
| dist/modern/index.html | 709.00b | +0.00b | 317.00ms | -73.00ms 🚀 |
| dist/legacy/styles.afb8e31a.css | 78.00b | +0.00b | 319.00ms | -72.00ms 🚀 |
| dist/modern/styles.cda8c91b.css | 78.00b | +0.00b | 317.00ms | -73.00ms 🚀 |

</p></details>

<details><summary>React HackerNews ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 7.41s | +157.00ms |
| Cached | 1.86s | -29.00ms |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/main/index.js | 66.80kb | +0.00b | 2.02s | +116.00ms ⚠️ |
| dist/module/index.js | 37.45kb | +0.00b | 2.02s | +119.00ms ⚠️ |

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/main/index.js | 66.80kb | +0.00b | 254.00ms | +31.00ms ⚠️ |
| dist/module/index.js | 37.45kb | +0.00b | 251.00ms | +26.00ms ⚠️ |

</p></details>

<details><summary>AtlasKit Editor ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 2.88m | -900.00ms |
| Cached | 4.82s | +690.00ms ⚠️ |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/EmojiPickerComponent.487f0492.js | 98.73kb | +0.00b | 22.27s | -2.01s 🚀 |
| dist/Modal.38c5bbd2.js | 44.27kb | +0.00b | 4.22s | +371.00ms ⚠️ |
| dist/png-chunks-extract.883f0434.js | 25.63kb | +0.00b | 9.03s | +1.53s ⚠️ |
| dist/DatePicker.00ddd357.js | 21.63kb | +0.00b | 16.86s | -3.74s 🚀 |
| dist/js.786dad09.js | 16.49kb | +0.00b | 4.17s | -5.15s 🚀 |
| dist/workerHasher.8c8d4944.js | 11.69kb | +0.00b | 22.98s | +2.07s ⚠️ |
| dist/16.d3236fb9.js | 10.22kb | +0.00b | 4.17s | +349.00ms ⚠️ |
| dist/16.c42426b9.js | 10.09kb | +0.00b | 4.17s | +349.00ms ⚠️ |
| dist/16.bc89f6ee.js | 10.02kb | +0.00b | 4.17s | +349.00ms ⚠️ |
| dist/16.98f3baa0.js | 9.98kb | +0.00b | 8.50s | +4.68s ⚠️ |
| dist/date.1c037d1c.js | 9.04kb | +0.00b | 9.83s | -1.70s 🚀 |
| dist/images.92799093.js | 8.98kb | +0.00b | 10.43s | -769.00ms 🚀 |
| dist/feedback.bffa274a.js | 8.95kb | +0.00b | 16.85s | +2.51s ⚠️ |
| dist/status.97878d50.js | 8.77kb | +0.00b | 1.05m | +50.55s ⚠️ |
| dist/link.c9af7470.js | 8.61kb | +0.00b | 11.12s | -51.87s 🚀 |
| dist/heading6.6daffbba.js | 8.61kb | +0.00b | 16.85s | +2.52s ⚠️ |
| dist/emoji.8c2bc3de.js | 8.54kb | +0.00b | 11.29s | +805.00ms ⚠️ |
| dist/expand.1a4c120f.js | 8.46kb | +0.00b | 16.85s | +2.51s ⚠️ |
| dist/divider.acfaa3b9.js | 8.33kb | +0.00b | 11.29s | -590.00ms 🚀 |
| dist/quote.55197fd6.js | 8.33kb | +0.00b | 12.24s | -51.17s 🚀 |
| dist/decision.5cdc67b3.js | 8.29kb | +0.00b | 9.83s | -1.76s 🚀 |
| dist/panel-error.8bc896be.js | 8.19kb | +0.00b | 1.05m | +51.03s ⚠️ |
| dist/panel.6acb9fb3.js | 8.18kb | +0.00b | 12.24s | -51.17s 🚀 |
| dist/panel-note.2eea2289.js | 8.13kb | +0.00b | 1.05m | +51.03s ⚠️ |
| dist/16.7f86346a.js | 4.89kb | +0.00b | 7.89s | -437.00ms 🚀 |
| dist/16.c81c0695.js | 4.82kb | +0.00b | 10.98s | +1.06s ⚠️ |
| dist/16.232fb9da.js | 4.69kb | +0.00b | 11.06s | +1.14s ⚠️ |
| dist/16.60895c0e.js | 4.65kb | +0.00b | 7.89s | -437.00ms 🚀 |
| dist/16.771934f3.js | 4.65kb | +0.00b | 7.18s | -2.84s 🚀 |
| dist/16.8a21dead.js | 4.64kb | +0.00b | 7.89s | -437.00ms 🚀 |
| dist/16.ba7ed30e.js | 4.64kb | +0.00b | 7.89s | -437.00ms 🚀 |
| dist/ak-editor.47799eae.css | 3.43kb | +0.00b | 1.73m | +1.67m ⚠️ |
| dist/ResourcedEmojiComponent.4495d16e.js | 2.32kb | +0.00b | 8.50s | -622.00ms 🚀 |
| dist/workerHasher.853c799b.js | 1.67kb | +0.00b | 22.97s | +2.06s ⚠️ |
| dist/media-viewer-analytics-error-boundary.62b79dc4.js | 1.59kb | +0.00b | 30.01s | -3.59s 🚀 |
| dist/simpleHasher.3db92b55.js | 642.00b | +0.00b | 22.98s | -1.34s 🚀 |
| dist/index.html | 212.00b | +0.00b | 943.00ms | -5.96s 🚀 |

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/ak-editor.255de60c.js | 2.49mb | +0.00b | 602.00ms | +68.00ms ⚠️ |
| dist/pdfRenderer.7bec20fe.js | 1.13mb | +0.00b | 595.00ms | +63.00ms ⚠️ |
| dist/editorView.4303a38b.js | 586.94kb | +0.00b | 595.00ms | +64.00ms ⚠️ |
| dist/popup.65f42bc5.js | 218.07kb | +0.00b | 595.00ms | +63.00ms ⚠️ |
| dist/esm.672cd889.js | 138.03kb | +0.00b | 600.00ms | +68.00ms ⚠️ |
| dist/Toolbar.a6408d78.js | 100.36kb | +0.00b | 595.00ms | +64.00ms ⚠️ |
| dist/EmojiPickerComponent.487f0492.js | 98.73kb | +0.00b | 600.00ms | +68.00ms ⚠️ |
| dist/media-viewer.30301157.js | 71.17kb | +0.00b | 600.00ms | +68.00ms ⚠️ |
| dist/card.d03c4017.js | 67.48kb | +0.00b | 600.00ms | +68.00ms ⚠️ |
| dist/ui.7b68b90d.js | 55.44kb | +0.00b | 600.00ms | +68.00ms ⚠️ |
| dist/smartMediaEditor.80585ed9.js | 55.41kb | +0.00b | 595.00ms | +64.00ms ⚠️ |
| dist/component.e0d6b61f.js | 52.79kb | +0.00b | 602.00ms | +69.00ms ⚠️ |
| dist/Modal.38c5bbd2.js | 44.27kb | +0.00b | 602.00ms | +68.00ms ⚠️ |
| dist/component.fd1c4fbd.js | 41.81kb | +0.00b | 602.00ms | +68.00ms ⚠️ |
| dist/png-chunks-extract.883f0434.js | 25.63kb | +0.00b | 602.00ms | +69.00ms ⚠️ |
| dist/EmojiPickerComponent.bd6f5b65.js | 22.42kb | +0.00b | 600.00ms | +68.00ms ⚠️ |
| dist/DatePicker.00ddd357.js | 21.63kb | +0.00b | 600.00ms | +68.00ms ⚠️ |
| dist/dropzone.b3681221.js | 17.75kb | +0.00b | 595.00ms | +64.00ms ⚠️ |
| dist/clipboard.d7548227.js | 17.21kb | +0.00b | 595.00ms | +64.00ms ⚠️ |
| dist/js.786dad09.js | 16.49kb | +0.00b | 602.00ms | +68.00ms ⚠️ |
| dist/browser.662842af.js | 16.08kb | +0.00b | 595.00ms | +64.00ms ⚠️ |
| dist/workerHasher.8c8d4944.js | 11.69kb | +0.00b | 600.00ms | +68.00ms ⚠️ |
| dist/16.d3236fb9.js | 10.22kb | +0.00b | 602.00ms | +68.00ms ⚠️ |
| dist/16.c42426b9.js | 10.09kb | +0.00b | 602.00ms | +68.00ms ⚠️ |
| dist/16.bc89f6ee.js | 10.02kb | +0.00b | 602.00ms | +68.00ms ⚠️ |
| dist/16.98f3baa0.js | 9.98kb | +0.00b | 602.00ms | +68.00ms ⚠️ |
| dist/EmojiTypeAheadComponent.a96819ac.js | 9.31kb | +0.00b | 600.00ms | +68.00ms ⚠️ |
| dist/date.1c037d1c.js | 9.04kb | +0.00b | 602.00ms | +69.00ms ⚠️ |
| dist/images.92799093.js | 8.98kb | +0.00b | 601.00ms | +68.00ms ⚠️ |
| dist/feedback.bffa274a.js | 8.95kb | +0.00b | 600.00ms | +68.00ms ⚠️ |
| dist/fallback.aab3a799.js | 8.90kb | +0.00b | 601.00ms | +68.00ms ⚠️ |
| dist/list-number.1316e9f7.js | 8.77kb | +0.00b | 601.00ms | +68.00ms ⚠️ |
| dist/status.97878d50.js | 8.77kb | +0.00b | 601.00ms | +68.00ms ⚠️ |
| dist/code.6b50d9e2.js | 8.69kb | +0.00b | 602.00ms | +69.00ms ⚠️ |
| dist/link.c9af7470.js | 8.61kb | +0.00b | 601.00ms | +69.00ms ⚠️ |
| dist/heading6.6daffbba.js | 8.61kb | +0.00b | 600.00ms | +68.00ms ⚠️ |
| dist/heading3.bcae1464.js | 8.60kb | +0.00b | 601.00ms | +68.00ms ⚠️ |
| dist/emoji.8c2bc3de.js | 8.54kb | +0.00b | 601.00ms | +68.00ms ⚠️ |
| dist/heading5.6844d092.js | 8.48kb | +0.00b | 601.00ms | +69.00ms ⚠️ |
| dist/expand.1a4c120f.js | 8.46kb | +0.00b | 600.00ms | +68.00ms ⚠️ |
| dist/heading2.a19338b9.js | 8.42kb | +0.00b | 601.00ms | +68.00ms ⚠️ |
| dist/mention.a261500a.js | 8.38kb | +0.00b | 601.00ms | +68.00ms ⚠️ |
| dist/heading4.02eb6187.js | 8.37kb | +0.00b | 601.00ms | +69.00ms ⚠️ |
| dist/layout.524db8d2.js | 8.36kb | +0.00b | 601.00ms | +68.00ms ⚠️ |
| dist/divider.acfaa3b9.js | 8.33kb | +0.00b | 601.00ms | +68.00ms ⚠️ |
| dist/quote.55197fd6.js | 8.33kb | +0.00b | 601.00ms | +68.00ms ⚠️ |
| dist/action.536bde1c.js | 8.31kb | +0.00b | 602.00ms | +69.00ms ⚠️ |
| dist/decision.5cdc67b3.js | 8.29kb | +0.00b | 602.00ms | +69.00ms ⚠️ |
| dist/panel-warning.cb245541.js | 8.29kb | +0.00b | 601.00ms | +69.00ms ⚠️ |
| dist/list.624b775a.js | 8.26kb | +0.00b | 601.00ms | +68.00ms ⚠️ |
| dist/heading1.ef0290b4.js | 8.26kb | +0.00b | 601.00ms | +68.00ms ⚠️ |
| dist/panel-error.8bc896be.js | 8.19kb | +0.00b | 601.00ms | +69.00ms ⚠️ |
| dist/panel.6acb9fb3.js | 8.18kb | +0.00b | 601.00ms | +68.00ms ⚠️ |
| dist/table.500adb35.js | 8.18kb | +0.00b | 601.00ms | +68.00ms ⚠️ |
| dist/panel-success.7636d22e.js | 8.13kb | +0.00b | 601.00ms | +69.00ms ⚠️ |
| dist/panel-note.2eea2289.js | 8.13kb | +0.00b | 601.00ms | +69.00ms ⚠️ |
| dist/16.073347bf.js | 5.24kb | +0.00b | 602.00ms | +69.00ms ⚠️ |
| dist/16.c4ee6c00.js | 5.17kb | +0.00b | 602.00ms | +69.00ms ⚠️ |
| dist/16.7f86346a.js | 4.89kb | +0.00b | 602.00ms | +69.00ms ⚠️ |
| dist/16.5f13de00.js | 4.84kb | +0.00b | 601.00ms | +68.00ms ⚠️ |
| dist/16.51c4a0ba.js | 4.82kb | +0.00b | 602.00ms | +69.00ms ⚠️ |
| dist/16.c81c0695.js | 4.82kb | +0.00b | 602.00ms | +69.00ms ⚠️ |
| dist/16.61fcf150.js | 4.78kb | +0.00b | 602.00ms | +69.00ms ⚠️ |
| dist/16.7b85ceeb.js | 4.73kb | +0.00b | 602.00ms | +69.00ms ⚠️ |
| dist/16.ba932c2e.js | 4.70kb | +0.00b | 602.00ms | +69.00ms ⚠️ |
| dist/16.232fb9da.js | 4.69kb | +0.00b | 602.00ms | +69.00ms ⚠️ |
| dist/16.60895c0e.js | 4.65kb | +0.00b | 602.00ms | +69.00ms ⚠️ |
| dist/16.771934f3.js | 4.65kb | +0.00b | 602.00ms | +69.00ms ⚠️ |
| dist/16.8a21dead.js | 4.64kb | +0.00b | 602.00ms | +69.00ms ⚠️ |
| dist/16.ba7ed30e.js | 4.64kb | +0.00b | 602.00ms | +69.00ms ⚠️ |
| dist/16.ea6e7d4b.js | 4.63kb | +0.00b | 602.00ms | +69.00ms ⚠️ |
| dist/16.ba0c7d86.js | 4.56kb | +0.00b | 602.00ms | +69.00ms ⚠️ |
| dist/EmojiPickerComponent.11d65c6e.js | 3.72kb | +0.00b | 600.00ms | +68.00ms ⚠️ |
| dist/ak-editor.47799eae.css | 3.43kb | +0.00b | 595.00ms | +64.00ms ⚠️ |
| dist/EmojiUploadComponent.b1f515dc.js | 2.53kb | +0.00b | 600.00ms | +68.00ms ⚠️ |
| dist/ResourcedEmojiComponent.4495d16e.js | 2.32kb | +0.00b | 602.00ms | +68.00ms ⚠️ |
| dist/workerHasher.853c799b.js | 1.67kb | +0.00b | 600.00ms | +68.00ms ⚠️ |
| dist/media-picker-analytics-error-boundary.17cad0d5.js | 1.63kb | +0.00b | 595.00ms | +64.00ms ⚠️ |
| dist/media-card-analytics-error-boundary.030a93f8.js | 1.62kb | +0.00b | 595.00ms | +63.00ms ⚠️ |
| dist/media-viewer-analytics-error-boundary.62b79dc4.js | 1.59kb | +0.00b | 595.00ms | +63.00ms ⚠️ |
| dist/simpleHasher.3db92b55.js | 642.00b | +0.00b | 600.00ms | +68.00ms ⚠️ |
| dist/EmojiPickerComponent.39e7a3dc.js | 467.00b | +0.00b | 600.00ms | +68.00ms ⚠️ |
| dist/index.html | 212.00b | +0.00b | 615.00ms | +80.00ms ⚠️ |

</p></details>

<details><summary>Three.js x4 🚨</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | **FAILED** | -0.00ms |
| Cached | **FAILED** | -0.00ms |

#### Cold Bundles

*No bundles found, this is probably a failed build...*
#### Cached Bundles

*No bundles found, this is probably a failed build...*
</p></details>

**[Click here to view a detailed benchmark overview.](https://benchmark.parceljs.org/benchmark/263360823569678858)**

File: packages/core/types/index.js [hunk 9]
```diff
@@ -300,6 +399,30 @@ export type ASTGenerator = {|
...
 |};
 
+/**
+ * An asset (usually represents one source file).
+ *
+ * @property The file system where the source is located.
+ * @property isIsolated Whether this asset should be put in a separate bundle. VERIFY
+ * @property isInline Whether this asset will/should later be inserted back into the importer.
+ * @property isSplittable FIXME
+ * @property isSource Whether this is asset is part of the users project (and not of an external dependencies).
```
Comment (mischnic) [inline packages/core/types/index.js:409]: TODO:
> `isSource` === "precompiled" (e.g. don't run Babel)

File: packages/core/types/index.js [hunk 10]
```diff
@@ -441,7 +599,10 @@ export type TransformerResult = {|
   +meta?: Meta,
   +pipeline?: ?string,
   +sideEffects?: boolean,
-  +symbols?: $ReadOnlyMap<Symbol, {|local: Symbol, loc: ?SourceLocation|}>,
+  +symbols?: $ReadOnlyMap<
+    CodeSymbol,
+    {|local: CodeSymbol, loc: ?SourceLocation|},
+  >,
```
Comment (mischnic) [inline packages/core/types/index.js:605]: Remove `symbolsConfident` below, it's unused

**Pull request closed without merging**